cmd/docker/builder.go

```diff
@@ -13,44 +12,29 @@ import (
 )
 
 const (
-	builderDefaultPlugin     = "buildx"
-	builderDefaultInstallMsg = `To install buildx, see https://docs.docker.com/go/buildx.`
-	builderErrorMsg          = `%s: Required builder component %s is missing or broken.`
-)
-
-type builderError struct {
-	warn    bool
-	builder string
-	err     error
-}
+	builderDefaultPlugin = "buildx"
+	buildxMissingWarning = `DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
+            Install the buildx component to build images with BuildKit:
```

Comment:
```suggestion
            Install the %s component to build images with BuildKit:
```

We have to specify the builder name dynamically in case a user use the non-default (buildx) one.

Reply:
You mean `buildx` without the `docker-` prefix? Isn't that a "incorrect" installation (i.e.; rename the binary)?

Reply:
If someone has an alias builder in his configuration we should use it. So if the alias builder is missing or broken that's the one we should display in the warning/error msg.

Reply:
So; 

- the `aliases` option was added specifically for buildx (to allow it to override the built-in `docker build` (https://github.com/docker/cli/pull/1840), and [not documented](https://docs.docker.com/engine/reference/commandline/cli/#cli-plugin-options) - deliberately as it was considered a temporary "hack" at the time. It was not the intent to allow `docker build` to be replaced with `<insert any other plugin>`
- even _if_ the aliases option would be kept, we'd still be only interested in `buildx`, not in any other plugin that's configured to replace `docker build` I don't think it makes sense to provide this error for `<insert some plugin>` (install the `kaniko` component to build images with BuildKit?) (the intent of the `buildx by default` change is to provide a consistent experience; 
- because of the above, I think we should consider removing the aliases option altogether, as the reason it was added (replace the built-in BuildKit client with the buildx one) is now the default, so we no longer need the option. (that is: remove the configuration option (or ignore it); if needed, we can keep the code logic in case other commands would be moved out of the main CLI (although there's no plans for that))

Reply:
This is just internal of how the plugins are loaded. We don't want to confuse users with this. The text should just say "buildx".

Reply:
> * the `aliases` option was added specifically for buildx (to allow it to override the built-in `docker build` (

Ah ok that makes sense, thanks!

Reply:
Do you think we should deprecate this `aliases` in the future?

Reply:
> Do you think we should deprecate the aliases in the future?

It's fine as a way to organize things and leaves more options for the future. But users should not worry about these internals.